e2e/src/componentsController.spec.ts

```diff
@@ -0,0 +1,639 @@
+import { createClient, createInvalidClient } from './config';
+import { uid } from 'uid';
+
+describe('Components Controller', () => {
+  let productFamily: ProductFamily | null;
+  let componentsController: ComponentsController;
+  let productsController: ProductsController;
+  let invalidComponentsController: ComponentsController;
+
+  beforeAll(async () => {
+    const client = createClient();
+    const invalidClient = createInvalidClient();
+    const productFamiliesController = new ProductFamiliesController(client);
+    invalidComponentsController = new ComponentsController(invalidClient);
+    componentsController = new ComponentsController(client);
+    productsController = new ProductsController(client);
+    productFamily =
+      (
+        await productFamiliesController.createProductFamily({
+          productFamily: {
+            name: `coupons-product-family-${uid()}`,
+            description: 'product family',
+          },
+        })
+      ).result.productFamily || null;
+  });
+
+  const invalidComponentId = 'invalid-id';
+
+  const prepareDataForComponentResponse = async (
+    productPaload: CreateOrUpdateProductRequest,
+    productFamilyId: number,
+    componentPayload: CreateMeteredComponent
+  ) => {
+    await productsController.createProduct(
+      productFamilyId.toString(),
+      productPaload
+    );
+
+    const createComponentReponse =
+      await componentsController.createMeteredComponent(productFamilyId, {
+        ...componentPayload,
+      });
+
+    return createComponentReponse.result;
+  };
+
+  describe('Find Component', () => {
+    test('should find component with valid id', async () => {
+      const newProductPayload: CreateOrUpdateProductRequest = {
+        product: {
+          name: `product-read-component-name-${uid()}`,
+          handle: `product-read-component-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+
+      const productFamilyId = productFamily?.id || 0;
+
+      const meteredComponent: CreateMeteredComponent = {
+        meteredComponent: {
+          productFamilyId,
+          name: `component-name-valid${uid()}`,
+          description: 'test',
+          quantity: 1,
+          unitName: 'test',
+          price: 1,
+          handle: uid(),
+          pricingScheme: PricingScheme.PerUnit,
+          unitPrice: 1,
+        },
+      };
+
+      const { component } = await prepareDataForComponentResponse(
+        newProductPayload,
+        productFamilyId,
+        meteredComponent
+      );
+
+      const componentHandle = component.handle || '';
+
+      const componentsControllerResponse =
+        await componentsController.findComponent(componentHandle);
+
+      expect(componentsControllerResponse.statusCode).toBe(200);
+      expect(componentsControllerResponse.result.component?.handle).toBe(
```

Comment:
We should add assertions on properties returned to ensure everything is correctly deserialized.